### PR TITLE
Feature/dropdown new style

### LINKED ISSUE #19: Dropdown multiselect (new)
**Decision from the Design review May 8**

The dropdown now has another version that allows multiple selections.

image

**Full specification**
See states and sizes here: https://designlibrary.sebgroup.com/components/forms/dropdown/#styles

**Component identifier**
component-dropdown

Thank you! :)
Ulrika
Design management

### PR DESCRIPTION
**Pull request by mario-subo**
This PR closes #19 

*NOTE: `react-select` can now be removed from dependencies*

How the old style compares to the new in a form:

Old style:
image

New style:
image

### CODE REVIEW

File: src/DropDown/DropDown.tsx [hunk 1]
```diff
@@ -185,14 +181,14 @@ export const DropDown: React.FunctionComponent<DropDownProps> = (props: DropDown
     // NATIVE ONCLICK EVENTS ================================
     /** The native event function that runs when a keyboard button is pressed on dropdown toggle */
     const handleKeyDownToggle = (event: React.KeyboardEvent<any>): void => {
-        const keyCode = event.which || event.keyCode;
-        if (keyCode !== 9 && keyCode !== 16) { // NOT tab or shift + tab
+        const key = event.key;
+        if (key !== "Tab" && !(event.shiftKey && key === "Tab")) { // NOT tab or shift + tab
```
[the remark this answers is not in the document]
Comment (DanSallau) [reply src/DropDown/DropDown.tsx]: @mario-subo i would recommend chaging tab to lowerCase first before casting them for assertion. I mean preferably 

```
const key = event.key.toLowerCase();
if (key !== "tab" && !(event.shiftKey && key === "tab"))
```

This way, you have a little control over the keyboard device event regardless if you miss a Case in the spelling or not.

File: src/DropDown/DropDown.tsx [hunk 2]
```diff
@@ -365,12 +361,12 @@ export const DropDown: React.FunctionComponent<DropDownProps> = (props: DropDown
 
                             <div className="right-items">
                                 {((props.clearable || props.multi) && selectedList.length > 0) ?
-                                    <div id="clearButton" className="icon-holder" onClick={shouldDisable ? null : handleClickClear}>{timesIcon}</div>
+                                    <div id="clearButton" className="custom-icon-holder" onClick={shouldDisable ? null : handleClickClear}>{timesIcon}</div>
```
[the remark this answers is not in the document]
Comment (mario-subo) [reply src/DropDown/DropDown.tsx]: resolved in commit https://github.com/sebgroup/react-components/pull/26/commits/a957ca1df2a0ae8732aa57d77362434be7c6f0f2

File: src/DropDown/DropDown.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (mario-subo) [reply src/DropDown/DropDown.tsx]: resolved in commit https://github.com/sebgroup/react-components/pull/26/commits/a957ca1df2a0ae8732aa57d77362434be7c6f0f2

File: src/DropDown/dropdown-style.scss [hunk 3]
```diff
@@ -1,94 +1,313 @@
+$transition-time: 300ms;
+$icon-size: 18px;
+$dropdown-width: 100%;
+$dropdown-toggle-height: 44px;
+$blue: #41b0ee;
+$darkBlue1: #0092e1;
+$darkBlue2: #007ac7;
+$text: #444444;
+$grey6: #999999;
+$grey7: #b3b3b3;
```
Comment (yousifalraheem) [inline src/DropDown/dropdown-style.scss]: Please change these to match SEB style variables

Comment (mario-subo) [reply src/DropDown/dropdown-style.scss]: fixed this in commit https://github.com/sebgroup/react-components/pull/26/commits/92ad13f895d88dc53f26f638dff9cd47c82d9999

File: src/DropDown/DropDown.tsx [hunk 4]
```diff
@@ -1,71 +1,460 @@
...
+            setInitialFocus();
+        }
+    };
+
+    const focusCurrentItem = (): boolean => {
+        if (shouldFocus && listRefs[currentFocused] && listRefs[currentFocused].current) {
+            listRefs[currentFocused].current.focus();
+            return true;
+        }
+        return null;
```
Comment (omarVodiaK) [inline src/DropDown/DropDown.tsx]: would it be fine to return null if this function type is boolean ?

File: src/DropDown/dropdown-style.scss [hunk 5]
```diff
@@ -1,94 +1,311 @@
...
+            &:active {
+                color: $body-color;
+                border-color: $blue-darker;
+                box-shadow: none;
+                background-color: white;
+            }
+        }
+
+        &.more {
+            background-color: inherit !important;
```
Comment (omarVodiaK) [inline src/DropDown/dropdown-style.scss]: is it possible to not use !important ?

Comment (mario-subo) [reply src/DropDown/dropdown-style.scss]: resolved in commit https://github.com/sebgroup/react-components/pull/26/commits/90ff6df057917ce882a2c165a3719737dfcc0a65

File: src/DropDown/DropDown.tsx (hunk 4 above)
Comment (mario-subo) [reply src/DropDown/DropDown.tsx]: resolved in commit https://github.com/sebgroup/react-components/pull/26/commits/90ff6df057917ce882a2c165a3719737dfcc0a65

**Pull request merged**